### PR TITLE
French translation of file menu checked (except for variances of Import Dataset menu

### PR DESCRIPTION
**Pull request by rachelkg**
I have looked through all the dialogs in this menu and updated them to make sure that the French text is always fully visible and correct. I have kept an excel file recording translation issues that need to be addressed through crowdin. I have not completed the Import Dataset dialog as I am waiting for further information. 
[R-Instat translation changes.xlsx](https://github.com/africanmathsinitiative/R-Instat/files/6711701/R-Instat.translation.changes.xlsx)

### CODE REVIEW

**Review by lloyddewit — DISMISSED** [review]

**Comment by lloyddewit** [comment]
@rdstern please can you test? thanks

**Comment by rdstern** [comment]
I copy @Ivanluv also on this issue, because I am finding some programming problems with the switching to French.  They may become a separate pull request from @Ivanluv or @lloyddewit 
I tried the first dialogue - in English - up to this option.  Then I went into French and re-opened the last dialogue.  I got this:
image

You can see from the bottom control that I am sort of in French, because you don't get the Lan button when only in English.

Please also note that the controls ate the bottom look "messy", i.e. the boxes don't all have the bottom line.  Is that a general problem, or just an alignment to be fixed by @rachelkg on each individual dialogue when it occurs?

But, more important is that you can see the text is still in English.  If I am in French, and switch back to English then there is the same problem. i.e. the text remains in French.  

Closing the dialogue and opening, does no good.  But if I hit another radio button at the top and then return it is fine, as shown:

image

So now some points for Rachel on this pull request:
a) First "Data Frame" itself.  Is it tableau de donnees or trame de donnees.  This is one we need to get right?
b) In the 4th option in English we have Rows: and Columns:  In French they lack the colon - I assume that is part of the  crowdin?  They are also singular in French.
c) Those boxes for the rows and columns don't quite line up - in either language.  Can the Rows move so the right-hand-side of the box lines up.  Then make the rows a bit wider, so it can show at least 100000.
d) In the other options Construction and Command, the Button for the Examples isn't big enough for the French.  It is ok for the Random (Aleatoire) option.

More later!

**Comment by rachelkg** [comment]
@rdstern Thanks for the feedback. I will look all this over tomorrow and get back to you.

**Comment by rachelkg** [comment]
@rdstern I believe I have successfully addressed all your comments on this.

**Review by rdstern — APPROVED** [review]
@shadrackkibet I am approving all 3 from @rachelkg as they are improvements and I hope that they can do no harm by being merged.  They will also allow  @berylwaswa and team to examine them for Swahili.
Can you please check the additions do no harm, and then hopefully merge.

**Review by lloyddewit — DISMISSED** [review]

**Review by lloyddewit — APPROVED** [review]

**Comment by lloyddewit** [comment]
@rdstern I removed some unecessary handler subs that were accidentally added to this PR. Please could you approve again so that we can merge?

@shadrackkibet @N-thony I removed the unnecessary handler subs by rolling back the affected files to the state they were in when this PR was first branched. This effectively removes the changed files from this PR. In case you ever need to do anything similar, I added my personal notes below:

- In Visual Studio
    - Ensure that you are in local master
    - Go to Git changes tab
        - Pull from upstream (to ensure that Master has the latest commits)
        - Note: you should normally also push to origin at this point to keep your GitHub fork up-to-date
    - Go to GitHub tab
        - Checkout the PR
    - Go to solution explorer tab
        - RMB on solution file, select open in file explorer
- In file explorer 
    - RMB Git Bash from here
- In Git Bash
    - Find the commit that the PR branch forked from
        - $ git merge-base --fork-point master
        - ace8d37e6665513f12a7afc0c70b1c2185b4af20
    - Check out, from the fork commit, all the files that you wish to roll back
        - $ git checkout ace8d37e6665513f12a7afc0c70b1c2185b4af20 -- dlgImportDataset.vb dlgNewDataFrame.vb dlgOpenNetCDF.vb dlgVisualizeData.vb ucrFilePath.vb
- In Visual Studio
    - Go to Git changes tab
        - Commit and push the changed files
- In GitHub
    - Go to the PR
    - Click on the files changed tab
    - The files you wanted to roll back should have disappeared from the list of changed files

File: instat/dlgExportRWorkspace.resx [hunk 1]
```diff
@@ -282,6 +282,14 @@
...
     <value>417, 335</value>
   </data>
+  <data name="ucrFilePath.FilePathBrowseText" xml:space="preserve">
+    <value>Unable to load DLL 'SQLite.Interop.dll': The specified module could not be found. (Exception from HRESULT: 0x8007007E)
+A problem occured attempting to translate string 'Browse' to language en using database C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\Common7\IDE\translations\rInstatTranslations.db.</value>
+  </data>
+  <data name="ucrFilePath.FilePathLabel" xml:space="preserve">
+    <value>Unable to load DLL 'SQLite.Interop.dll': The specified module could not be found. (Exception from HRESULT: 0x8007007E)
+A problem occured attempting to translate string 'Export File:' to language en using database C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\Common7\IDE\translations\rInstatTranslations.db.</value>
+  </data>
```
Comment (shadrackkibet) [inline instat/dlgExportRWorkspace.resx:292]: @lloyddewit I noted that there is some problem with the loading of a `DLL` here.

**Comment by shadrackkibet** [comment]
> @rdstern I removed some unecessary handler subs that were accidentally added to this PR. Please could you approve again so that we can merge?
> 
> @shadrackkibet @N-thony I removed the unnecessary handler subs by rolling back the affected files to the state they were in when this PR was first branched. This effectively removes the changed files from this PR. In case you ever need to do anything similar, I added my personal notes below:
> 
> * In Visual Studio
>   
>   * Ensure that you are in local master
>   * Go to Git changes tab
>     
>     * Pull from upstream (to ensure that Master has the latest commits)
>     * Note: you should normally also push to origin at this point to keep your GitHub fork up-to-date
>   * Go to GitHub tab
>     
>     * Checkout the PR
>   * Go to solution explorer tab
>     
>     * RMB on solution file, select open in file explorer
> * In file explorer
>   
>   * RMB Git Bash from here
> * In Git Bash
>   
>   * Find the commit that the PR branch forked from
>     
>     * $ git merge-base --fork-point master
>     * [ace8d37](https://github.com/africanmathsinitiative/R-Instat/commit/ace8d37e6665513f12a7afc0c70b1c2185b4af20)
>   * Check out, from the fork commit, all the files that you wish to roll back
>     
>     * $ git checkout [ace8d37](https://github.com/africanmathsinitiative/R-Instat/commit/ace8d37e6665513f12a7afc0c70b1c2185b4af20) -- dlgImportDataset.vb dlgNewDataFrame.vb dlgOpenNetCDF.vb dlgVisualizeData.vb ucrFilePath.vb
> * In Visual Studio
>   
>   * Go to Git changes tab
>     
>     * Commit and push the changed files
> * In GitHub
>   
>   * Go to the PR
>   * Click on the files changed tab
>   * The files you wanted to roll back should have disappeared from the list of changed files

Thanks, @lloyddewit this is a very useful and efficient way to remove files from PR. This will come in handy when we have similar problems. Perhaps good to document these steps somewhere for developers.

**Review by shadrackkibet — APPROVED** [review]

**Comment by shadrackkibet** [comment]
@rdstern please re-test and re-approve if you are happy.

**Review by rdstern — APPROVED** [review]
Adding my approval to hope we canm make progress!

File: instat/dlgExportRWorkspace.resx (hunk 1 above)
Comment (lloyddewit) [reply instat/dlgExportRWorkspace.resx:292]: @shadrackkibet The missing DLL is a known problem. The developer needs to manually install a NuGet package. @Wycklife will document this in the developers' guide on the Wiki.

Comment (Wycklife) [reply instat/dlgExportRWorkspace.resx:292]: @lloyddewit I will ensure this is documented before the end of the day.

**Comment by lloyddewit** [comment]
> 
> Thanks, @lloyddewit this is a very useful and efficient way to remove files from PR. This will come in handy when we have similar problems. Perhaps good to document these steps somewhere for developers.

@shadrackkibet Good idea, I created [this Wiki page](https://github.com/africanmathsinitiative/R-Instat/wiki/GitHub-notes).

File: instat/dlgExportRWorkspace.resx (hunk 1 above)
Comment (lloyddewit) [reply instat/dlgExportRWorkspace.resx:292]: > 
> 
> @lloyddewit I will ensure this is documented before the end of the day.

@Wycklife Please could you clarify the status? thanks

**Pull request merged**